spyt-package/src/main/python/spyt/conf.py

```diff
@@ -26,6 +18,71 @@
+        version_conf["cluster_version"] = cluster_version
+        return update_inplace(self.global_conf, version_conf)  # TODO(alex-shishkin): Might cause undefined behaviour
+
+    def latest_ytserver_proxy_path(self, cluster_version):
+        if cluster_version:
+            return None
+        global_conf = self.global_conf
+        symlink_path = global_conf.get("ytserver_proxy_path")
+        if symlink_path is None:
+            return None
+        return get("{}&/@target_path".format(symlink_path), client=self.client)
+
+    def validate_cluster_version(self, spark_cluster_version):
+        if not exists(self._get_version_conf_path(spark_cluster_version), client=self.client):
+            raise RuntimeError("Unknown SPYT cluster version: {}. Available release versions are: {}".format(
+                spark_cluster_version, self.get_available_cluster_versions()
+            ))
+        spyt_minor_version = SpytVersion(SELF_VERSION).get_minor()
+        cluster_minor_version = SpytVersion(spark_cluster_version).get_minor()
+        if spyt_minor_version < cluster_minor_version:
+            logger.warning("You required SPYT version {} which is older than your local ytsaurus-spyt version {}."
+                           "Please update your local ytsaurus-spyt".format(spark_cluster_version, SELF_VERSION))
+
+    def get_available_cluster_versions(self):
+        subdirs = yt_list(self.conf_base_path.join(RELEASES_SUBDIR), client=self.client)
+        return [x for x in subdirs if x != "spark-launch-conf"]
+
+    def latest_compatible_spyt_version(self, version):
+        minor_version = SpytVersion(version).get_minor()
+        spyt_versions = self.get_available_spyt_versions()
+        compatible_spyt_versions = [x for x in spyt_versions if SpytVersion(x).get_minor() == minor_version]
+        if not compatible_spyt_versions:
+            raise RuntimeError(f"No compatible SPYT versions found for specified version {version}")
+        return max(compatible_spyt_versions, key=SpytVersion)
+
+    def get_available_spyt_versions(self):
+        return yt_list(self.spyt_base_path.join(RELEASES_SUBDIR), client=self.client)
+
+    def get_spark_distributive(self):
+        distrib_root = self.distrib_base_path.join(spark_version.replace('.', '/'))
+        distrib_root_contents = yt_list(distrib_root, client=self.client)
+        spark_tgz = [x for x in distrib_root_contents if x.endswith('.tgz')]
+        if len(spark_tgz) == 0:
+            raise RuntimeError(f"Spark {spark_version} tgz distributive doesn't exist "
+                               f"at path {distrib_root} on cluster {self.client.config['proxy']['url']}")
+        return (spark_tgz[0], distrib_root.join(spark_tgz[0]))
+
+    def _get_version_conf_path(self, cluster_version):
+        return self.conf_base_path.join(self._version_subdir(cluster_version)).join(cluster_version).join("spark-launch-conf")
+
+    def _version_subdir(self, version):
```

Comment:
This function can be embedded into `_get_version_conf_path` because it is the only place where it is used.